test/jdk/java/awt/Frame/FrameVisualTest.java

```diff
@@ -57,32 +56,39 @@ public class FrameVisualTest {
     public static void main(String[] args) throws Exception {
         gcs = GraphicsEnvironment.getLocalGraphicsEnvironment().getDefaultScreenDevice().getConfigurations();
         robot = new Robot();
-        robot.setAutoDelay(100);
+        robot.waitForIdle();
         try {
-            EventQueue.invokeAndWait(() -> {
-                createAndShowUI();
-            });
+            EventQueue.invokeAndWait(() -> createAndShowUI());
+
+            robot.waitForIdle();
             robot.delay(1000);
-            System.out.println("frames.length: "+frames.length);
+
+            System.out.println("frames.length: " + frames.length);
             for (index = 0; index < frames.length; index++) {
                 EventQueue.invokeAndWait(() -> {
                     p = frames[index].getLocation();
                     d = frames[index].getSize();
                 });
                 Rectangle rect = new Rectangle(p, d);
                 BufferedImage img = robot.createScreenCapture(rect);
+                robot.waitForIdle();
+                robot.delay(100);
                 if (chkImgBackgroundColor(img)) {
                     try {
-                        ImageIO.write(img, "png", new File("Frame_" + index + ".png"));
+                        ImageIO.write(img, "png", new File("Frame_" + index +
+                                ".png"));
                     } catch (IOException ignored) {}
-                    throw new RuntimeException("Frame visual test failed with non-white background color");
+                    throw new RuntimeException("Frame visual test failed " +
+                            "with non-white background color");
                 }
             }
         } finally {
             for (index = 0; index < frames.length; index++) {
                 EventQueue.invokeAndWait(() -> {
                     if (frames[index] != null) {
+                        frames[index].setVisible(false);
                         frames[index].dispose();
```

Comment:
Calling `setVisible(false)` before `dispose` is redundant.

Reply:
Removed, thanks.